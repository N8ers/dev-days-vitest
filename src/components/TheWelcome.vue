<template>
  <div>
    <h1>Dev Days Vitest!</h1>
    <p>can you select me??</p>
  </div>
</template>
