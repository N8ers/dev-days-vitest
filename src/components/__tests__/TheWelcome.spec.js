import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";

import TheWelcome from "@/components/TheWelcome.vue";

describe("TheWelcome.vue", () => {
  test("Component renders h1 content", () => {
    // find the h1 tag and check the text value
  });

  test("Component renders p content", () => {
    // find the p tag and check the text value
  });

  // A snapshot is a "picture" of what the rendered html looks like
  // a file will be made in the "@/components/__tests__/__snapshots__" directory
  test("Component snapshot", () => {
    // You can write assertions on the entire wrapper!
    // snapshot the value
  });

  // More on snapshots
  // When you make a change to the component, you may need to update a snapshot
  // Show how, and explain how to diff
});
