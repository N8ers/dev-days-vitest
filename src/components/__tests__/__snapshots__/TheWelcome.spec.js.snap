// Vitest Snapshot v1

exports[`TheWelcome.vue > Component snapshot 1`] = `
<div>
  <h1>Dev Days Vitest!</h1>
  <p>can you select me??</p>
</div>
`;
